pkg/cmd/variables/variables_test.go

```diff
@@ -109,6 +109,11 @@ func TestCmdVariables(t *testing.T) {
 		if name == "nested" {
 			o.Options.Repository = "cluster/print-go"
 		}
+		if name == "app_name_override" {
+			args := []string{"--app", "overridden-app-name"}
+			err := cmd.ParseFlags(args)
```

Comment:
I dont think this test is required, you are just testing whether cobra parse flags work (an external function) :thinking: 
Just args line is sufficient.

Reply:
do you want me to get rid of it? I wrote a new test, but it seemed redundant because this one already overrides `o.Options.Repository`
```
		if name == "nested" {
			o.Options.Repository = "cluster/print-go"
		}
```
FYI, [the jx edit tests](https://github.com/jenkins-x-plugins/jx-gitops/blob/8a28561730e720f373125b28680c6b454555b482/pkg/cmd/requirement/edit/edit_test.go#L82-L85) actually test some cobra args

Reply:
Just removing the line 
```
err := cmd.ParseFlags(args)
require.NoError(t, err, "failed to parse arguments %#v for test app_name_override", args)
```
is fine, you can add args only for ur test case.